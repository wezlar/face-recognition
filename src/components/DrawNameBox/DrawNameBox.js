import React from 'react';
import PropTypes from 'prop-types';

const boxStyle = {
  backgroundColor: 'blue',
  border: 'solid',
  borderColor: 'blue',
  marginTop: 0,
  color: '#fff',
};

export const DrawNameBox = (props = {}) => {
  const { match, index, width, height } = props;
  if (!match || !match[index]) {
    return null;
  };
  
  return (
    <p
      style={{
        ...boxStyle,
        width,
        transform: `translate(-3px, ${height}px)`,
      }}>
      {match[index]._label}
    </p>
  );
};

DrawNameBox.propTypes = {
  match: PropTypes.array,
  index: PropTypes.number,
  width: PropTypes.number,
  height: PropTypes.number,
};

export default DrawNameBox;
