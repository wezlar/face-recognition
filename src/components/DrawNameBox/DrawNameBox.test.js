// __tests__/fetch.js
import React from 'react';
import {
  render,
  cleanup,
} from '@testing-library/react';
// this adds custom jest matchers from jest-dom
import DrawNameBox from './DrawNameBox';

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);
describe(`Test DrawNameBox`, () => {
  test(`Test snapshot`, () => {
    const match = [
      {
        _label: 'Test Label 1',
      },
      {
        _label: 'Test Label 2',
      }
    ];
    
    // Arrange
    const { container, getByTestId, debug } = render(
      <DrawNameBox 
        match={match} 
        index={0}
        width={10}
        height={10} />
    );

    // snapshots work great with regular DOM nodes!
    expect(container.firstChild).toMatchSnapshot();
    expect(getByTestId('draw-name-box').textContent).toBe(match[0]._label);
    expect(getByTestId('draw-name-box').style.width).toBe('10px');
    expect(getByTestId('draw-name-box').style.transform).toBe('translate(-3px, 10px)');
  });
});