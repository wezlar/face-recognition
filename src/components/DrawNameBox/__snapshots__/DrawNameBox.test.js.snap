// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DrawNameBox Test snapshot 1`] = `
<p
  data-testid="draw-name-box"
  style="background-color: blue; border: solid; border-color: blue; margin-top: 0px; color: rgb(255, 255, 255); width: 10px; transform: translate(-3px, 10px);"
>
  Test Label 1
</p>
`;
